rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -927,4 +936,79 @@ public void removeThirdPartyCredential(String key) throws NotConnectedException,
             throw new PermissionException(e);
         }
     }
+
+    @Override
+    public Page<TaskId> getTaskIds(String taskTag, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedException, PermissionException {
+        RestPage<TaskStateData> page = null;
+        try {
+            page = restApi().getTaskStates(sid, from, to, mytasks, running, pending, finished, offset,
+                    limit);
+        } catch (NotConnectedRestException e) {
+            throw new NotConnectedException(e);
+        } catch (PermissionRestException e) {
+            throw new PermissionException(e);
+        }
+        List<TaskId> lTaskIds = new ArrayList<TaskId>(page.getList().size());
+        for (TaskStateData taskStateData : page.getList()) {
+            JobId jobId = new JobIdImpl(taskStateData.getTaskInfo().getJobId().getId(),
```

Comment:
Could introduce variable for `taskStateData.getTaskInfo()` in order to improve readability.